src/main/java/by/itech/lab/supplier/controller/UserController.java

```diff
@@ -71,18 +72,18 @@ public UserDto createUser(@Valid @RequestBody UserDto userDto) {
 
 
     @GetMapping(ApiConstants.URL_USERNAME_PARAMETER)
-    public UserDto getUseByUsername(@PathVariable String username) {
+    public UserDto getUserByUsername(@PathVariable String username) {
         log.debug("request to get User : {}", username);
         return userService.findByUsername(username);
     }
-    @AdminAccess
+
```

Comment:
so, anyone will be able to change the password to anyone?

Reply:
Each user is able to change his password

Reply:
here I see the code, which says that I as UserA able to change the password for UserB (or even admin) if I only know his ID

Reply:
i've added in service method check if user principal id is the same as in url

Reply:
its not necessary actions, you could use something like:
@PreAuthorize("principal.id == :id")
// not sure in syntax, but you can check how to do it correctly